XamlAnimatedGif/Animator.cs

```diff
@@ -327,7 +386,7 @@ private async Task RenderFrameAsync(int frameIndex, CancellationToken cancellati
 
                     for (int x = 0; x < rect.Width; x++)
                     {
-                        byte index = indexBuffer[x];
+                        byte index = _cacheFrameDataInMemory ? indexDecompressedBytes[y][x] : indexBuffer[x];
```

Comment:
What bothers me about this whole method is how the two way of accessing pixel data is interleaved. There are too many places where we need to check `_cacheFrameDataInMemory` to do something different, causing some variables to be initialized or not. It makes the method much more complex and harder to reason about.

Could you try to refactor this so that the `_cacheFrameDataInMemory` code path and the "normal" code path are clearly separate? It might mean duplicating some code, but at least it'll be easier to follow.

Reply:
I tried to do this. Now it only checks it twice, first to initialize the indexStream when reading from gif frames and then check it to read indexBuffer either from indexStream or from _cachedFrameBytes

Reply:
Much better, thanks!